Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLReader.java

```diff
@@ -0,0 +1,561 @@
+    }
+
+    /**
+     * Creates a Node from a file.
+     *
+     * @param path
+     *            file to read
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the linked data node
+     * @throws SAXException
+     *             if the file is not wellformed
+     * @throws IOException
+     *             if the reading fails
+     */
+    public static Node toNode(File path, Storage storage) throws SAXException, IOException {
+        return toNode(parseXML(new FileInputStream(path), Optional.empty()), uriForFile(path), storage);
+    }
+
+    /**
+     * Creates a Node from an arbitrary input resource.
+     *
+     * @param bytes
+     *            input data
+     * @param encoding
+     *            character encoding to use
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the linked data node
+     * @throws SAXException
+     *             if the file is not well-formed
+     * @throws IOException
+     *             if the reading fails
+     */
+    public static Node toNode(InputStream bytes, Optional<String> encoding, String documentNS, Storage storage)
+            throws SAXException, IOException {
+        return toNode(parseXML(bytes, encoding), documentNS, storage);
+    }
+
+    /**
+     * Create a Node from an XML String.
+     *
+     * @param data
+     *            String to parse
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the created Node
+     * @throws SAXException
+     *             if the XML is semantically wrong
+     */
+    public static Node toNode(String data, String documentNS, Storage storage) throws SAXException {
+        try {
+            return toNode(new ByteArrayInputStream(data.getBytes("UTF-16")), Optional.of("UTF-16"), documentNS,
+                storage);
+        } catch (UnsupportedEncodingException e) {
+            /* "UTF-16" is always supported by Java */
+            throw new IllegalStateException(e.getMessage(), e);
+        } catch (IOException e) {
+            /* there is no IOException to expect when reading from a String */
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Returns an URI for a local file, with the host name of the machine the
+     * program is running on.
+     *
+     * @param file
+     *            file to create an URI for
+     * @return URI for file
+     * @throws IOException
+     *             if it fails
+     */
+    public static String uriForFile(File file) throws IOException {
```

Comment:
When I read this method then I would expect and URI object is returned but you are returning a String object? I think a URI object is a better solution or adjust the JavaDoc and method name.